backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/sequence/BillToCountryIdProvider.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2023 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.document.sequence;
+
+import de.metas.location.CountryId;
+import lombok.NonNull;
+import lombok.Value;
+import org.compiere.util.Evaluatee;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+public interface BillToCountryIdProvider
+{
+	@Value
```

Comment:
Later addition (related to my comment in the SQL): maybe also declare a method like `isHandled(DocBaseAndSubType)`